### DISCUSSION TITLE
namespaces not showing graphs in v.1.31 and higher

### DISCUSSION THREAD

**Discussion by frankthing** [kiali/kiali#4102] [Q&A]
Hi! I'm quite new to kiali, and i have been struggling with a issue that you guys might have an answer to: 

When running v.130 everything works as expected regarding thew graph view, and as you can see in this screenshot, graph is showing ok when filtered to only the workspace "helloworld".
![image](https://user-images.githubusercontent.com/85570471/122064399-6e85b180-cdf1-11eb-9e29-43c0ec89ff07.png)

I am also able to filter on namespace istio-system to see the traffic to the services running in oth namespae helloworld and book-info. 

![image](https://user-images.githubusercontent.com/85570471/122064846-bf95a580-cdf1-11eb-8988-a824f681ed52.png)

However, when I update to v.1.31 or higher, I can't see any traffic when filtering on the namespaces other than istio-system. 
This is namespace helloworld (same thing for book-info) when running v1.32: 
![image](https://user-images.githubusercontent.com/85570471/122065268-17cca780-cdf2-11eb-9b40-47b0eff2c22f.png)

But if I filter on istio-system it shows the traffic. This is filtered on istio-system (v.1.32):
![image](https://user-images.githubusercontent.com/85570471/122065446-3f237480-cdf2-11eb-8734-49d3d1db359f.png)

Anyone have the same experience?

**Comment by jshaughn** [comment]
As an aside, Kiali 1.35 is now available, although I don't think it would change whatever you are seeing.  I'm not exactly sure why you would see this change of behavior.  The symptoms would maybe indicate that you don't have istio sidecars injected into your helloworld or bookinfo pods.  Do you have the Missing Sidecars Display option enabled?   Or, in the Workloads list view do you see any missing sidecars indicated?

**Reply by frankthing** [reply]
I have tried every version up to 1.35 with the same behaviour. No missing sidecars as far as I can see, it also works correctly after downgrading to 1.30, no other changes on the pods in helloworld or book-info.

**Reply by jshaughn** [reply]
Can you take a look at your kiali pod logs and see if you find anything unusual when trying to generate the graph?  Especially any indication that may show that metrics are being skipped?  You should be able to onspect the Kiali pod logs inside Kiali, via the Workload detail tabs.

Ift hat all looks fine then can you peek into your Prometheus metrics (see [this FAQ if you need help](https://kiali.io/documentation/latest/faq/#prometheus)).   Maybe capture the results of:

istio_requests_total{destination_service_name="helloworld"}

and post as an attachment.

**Reply by jshaughn** [reply]
Also, what is your Istio version?

**Reply by frankthing** [reply]
We are running Istio 1.9.1. 
Actually found the reason, we were deploying a default network policy on all our namespaces to isolate them network wise. Just had to allow prometheus to communicate across the namespaces. :)

**Reply by jshaughn** [reply]
Bad wiring makes sense, I couldn't really explain the behavior otherwise.  Thanks for letting us know.
